src/utils.ts

```diff
@@ -10,15 +10,27 @@ import { config } from "./config";
 const isWin = /^win/.test(process.platform);
 const dartExecutableName = isWin ? "dart.exe" : "dart";
 const pubExecutableName = isWin ? "pub.bat" : "pub";
+const flutterExecutableName = isWin ? "flutter.bat" : "flutter";
 export const dartVMPath = "bin/" + dartExecutableName;
 export const dartPubPath = "bin/" + pubExecutableName;
 export const analyzerPath = "bin/snapshots/analysis_server.dart.snapshot";
 export const extensionVersion = getExtensionVersion();
 export const isDevelopment = checkIsDevelopment();
 
+export function isFlutterProject(): boolean {
+	if (workspace.rootPath)
+		if (fs.existsSync(path.join(workspace.rootPath, "pubspec.yaml")))
+			return fs.readFileSync((path.join(workspace.rootPath, "pubspec.yaml"))).includes("sdk: flutter");
+	return false;
+}
+
 export function findDartSdk(): string {
 	let paths = (<string>process.env.PATH).split(path.delimiter);
 
+	// Putting Flutter before userDefined, so that UD takes priority
+	if (isFlutterProject)
```

Comment:
I'm not sure (I've never used Flutter), but I think it might be invalid/incorrect to use the "normal" Dart SDK for Flutter projects. If so, it might be best to export a new `findSdk` function which internally just calls `findDartSdk` or `findFlutterSdk` accordingly. This would also make it easier in the flutter case if `findFlutterSdk` doesn't find it to alert the user to do whatever is required to pull the flutter-dart-sdk down.